### PR TITLE
PWGHF: Add Bs workflow

### PR DESCRIPTION
**Pull request by pstahlhu**
First implementation of the Bs workflow @fcatalan92 @fgrosa

### CODE REVIEW

**Comment by vkucera** [comment]
Hi @pstahlhu , thanks for your contribution. I converted the PR to a draft for now, since some changes will be required after https://github.com/AliceO2Group/O2Physics/pull/3244 is merged, which should be done today.

File: PWGHF/DataModel/CandidateSelectionTables.h [hunk 1]
```diff
@@ -296,6 +296,102 @@ bool selectionPID(const T1& pidTrackPi, const T2& acceptPIDNotApplicable)
...
+template <typename T1, typename T2, typename T3>
+bool selectionTopol(const T1& candBs, const T2& cuts, const T3& binsPt)
+{
+  auto ptCandBs = candBs.pt();
+  auto ptDs = RecoDecay::pt(candBs.pxProng0(), candBs.pyProng0());
+  auto ptPi = RecoDecay::pt(candBs.pxProng1(), candBs.pyProng1());
+
+  int pTBin = findBin(binsPt, ptCandBs);
+  if (pTBin == -1) {
+    LOGF(info, "Bs topol selection failed at getpTBin");
```
Comment (fgrosa) [inline PWGHF/DataModel/CandidateSelectionTables.h:317]: I would remove this message since it could be repeated many times if your binning does not start from 0 (hence it is probable to get `-1` from the `findBin` function)
```suggestion
```

File: PWGHF/TableProducer/candidateCreatorBs.cxx [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    // Match generated particles.
+    for (auto const& particle : particlesMc) {
+      flag = 0;
+      origin = 0;
+      arrDaughDsIndex.clear();
+      // Bs(bar) → Ds∓ π±
+      if (RecoDecay::isMatchedMCGen(particlesMc, particle, pdg::Code::kBS, array{-static_cast<int>(pdg::Code::kDS), +kPiPlus}, true)) {
+        // Match Ds∓ -> K- K+ π±
+        auto candDsMC = particlesMc.rawIteratorAt(particle.daughtersIds().front());
+        if (RecoDecay::isMatchedMCGen(particlesMc, candDsMC, -static_cast<int>(pdg::Code::kDS), array{-kKPlus, +kKPlus, -kPiPlus}, true, &sign, 2)) {
```
Comment (fgrosa) [inline PWGHF/TableProducer/candidateCreatorBs.cxx:427]: Since you defined `pdg::Code::kDSBar` you can use it also here
```suggestion
        if (RecoDecay::isMatchedMCGen(particlesMc, candDsMC, pdg::Code::kDSBar, array{-kKPlus, +kKPlus, -kPiPlus}, true, &sign, 2)) {
```

**Review by fgrosa — COMMENTED** [review]
Hi @pstahlhu, thanks a lot for the work! I just have a couple of minor comments (see below).

**Review by fcatalan92 — COMMENTED** [review]
Hi @pstahlhu, thanks a lot for the very nice developments. I added some line-by-line comments/suggestions. Besides those, the PR looks good to me.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 3]
```diff
@@ -1897,18 +1897,15 @@ DECLARE_SOA_INDEX_COLUMN_FULL(Prong0, prong0, int, HfCand3Prong, "_0"); // Ds in
...
-DECLARE_SOA_COLUMN(OriginMcGen, originMcGen, int8_t);       // particle origin, generator level
-DECLARE_SOA_COLUMN(DebugMcRec, debugMcRec, int8_t);         // debug flag for mis-association reconstruction level
 
 // mapping of decay types
 enum DecayType { BsToDsPi };
 
-enum DecayTypeMc : uint8_t { BsToDsPiToKKPiPi = 0,
-                             PartlyRecoDecay,
-                             OtherDecay,
-                             NDecayTypeMc };
```
Comment (vkucera) [inline PWGHF/DataModel/CandidateReconstructionTables.h]: Why did you remove this?

File: PWGHF/D2H/Tasks/taskBs.cxx [hunk 4]
```diff
@@ -124,7 +124,7 @@ struct HfTaskBs {
     registry.add("hYGenWithProngsInAcceptance", "MC particles (generated-daughters in acceptance);B^{0}_{s} candidate #it{y}^{gen};entries", {HistType::kTH2F, {{100, -2., 2.}, axisPt}});
 
     if (checkDecayTypeMc) {
-      constexpr uint8_t kNBinsDecayTypeMc = DecayTypeMc::NDecayTypeMc;
+      constexpr uint8_t kNBinsDecayTypeMc = DecayTypeMc::OtherDecay;
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskBs.cxx]: This will make `kNBinsDecayTypeMc = 2` so `labels` will only have 2 elements and `labels[DecayTypeMc::OtherDecay] = "Other decays";` will crash.

**Review by fcatalan92 — APPROVED** [review]
Hi @pstahlhu, thanks a lot for implementing the comments. 
@vkucera  This PR is ready to be merged for me

**Review by vkucera — APPROVED** [review]
Thanks @pstahlhu for your contribution.
Thanks @fcatalan92 @fgrosa for your reviews.

**Comment by pstahlhu** [comment]
Thank you very much for your comments @vkucera @fcatalan92 @fgrosa!
